### PR TITLE
Convert ReactRenderDocument to hydrateRoot

### PR DESCRIPTION
**Pull request by rickhanlonii**

### CODE REVIEW

File: packages/react-dom/src/__tests__/ReactRenderDocument-test.js [hunk 1]
```diff
@@ -153,40 +173,80 @@ describe('rendering React components at document', () => {
...
+
+      expect(() => {
+        ReactDOM.flushSync(() => {
+          ReactDOMClient.hydrateRoot(container, <div>parsnip</div>, {
+            onRecoverableError: error => {
+              Scheduler.log('Log recoverable error: ' + error.message);
+            },
+          });
+        });
+      }).toErrorDev(
```
Comment (rickhanlonii) [inline packages/react-dom/src/__tests__/ReactRenderDocument-test.js:198]: These test have a bunch of improvements to the errors in the new root.

File: packages/react-dom/src/__tests__/ReactRenderDocument-test.js [hunk 2]
```diff
@@ -153,40 +173,80 @@ describe('rendering React components at document', () => {
...
       );
+
+      assertLog([
+        'Log recoverable error: Hydration failed because the initial UI does not match what was rendered on the server.',
+        'Log recoverable error: There was an error while hydrating. Because the error happened outside of a Suspense boundary, the entire root will switch to client rendering.',
+      ]);
+
       // This creates an unfortunate double text case.
-      expect(container.textContent).toBe('potatoparsnip');
+      expect(container.textContent).toBe('parsnip');
```
Comment (rickhanlonii) [inline packages/react-dom/src/__tests__/ReactRenderDocument-test.js:212]: Client render replaces the content with the client version.

File: packages/react-dom/src/__tests__/ReactRenderDocument-test.js [hunk 3]
```diff
@@ -229,23 +310,59 @@ describe('rendering React components at document', () => {
...
+        assertLog([
+          'Log recoverable error: Text content does not match server-rendered HTML.',
+          'Log recoverable error: There was an error while hydrating. Because the error happened outside of a Suspense boundary, the entire root will switch to client rendering.',
+        ]);
       }
       expect(testDocument.body.innerHTML).toBe('Hello world');
     });
 
-    it('supports findDOMNode on full-page components', () => {
+    it('supports findDOMNode on full-page components in legacy mode', () => {
```
Comment (rickhanlonii) [inline packages/react-dom/src/__tests__/ReactRenderDocument-test.js:365]: Specifically testing findDOMNode.

**Comment by eps1lon** [comment]
Seems like www is not happy with the new tests.

**Review by eps1lon — APPROVED** [review]

**Pull request merged**